### PR TITLE
Enabled Bad data logging for STP agent

### PR DESCRIPTION
**Pull request by nicklester**
Enabled bad data logging for STP agent in line with bad data logging for data tools app

### CODE REVIEW

File: server/cmwell-data-tools-app/src/main/resources/logback.xml [hunk 1]
```diff
@@ -107,15 +107,14 @@
 
     <appender name="ASYNC_BAD_DATA" class="ch.qos.logback.classic.AsyncAppender">
         <appender-ref ref="BAD_DATA" />
-        <maxFlushTime>0</maxFlushTime>
```
Comment (e-orz) [inline server/cmwell-data-tools-app/src/main/resources/logback.xml]: The assumption is that no log line should be discarded.
Why did you remove the unlimited time waited for log line to be printed?

File: server/cmwell-data-tools-app/src/main/resources/logback.xml [hunk 2]
```diff
@@ -107,15 +107,14 @@
...
         <discardingThreshold>0</discardingThreshold>
     </appender>
 
     <logger name="tools-red-logger" additivity="true">
         <appender-ref ref="ASYNC_RED_LOG" />
     </logger>
 
-    <logger name="bad-data" additivity="false">
+    <logger name="bad-data" additivity="true">
```
Comment (e-orz) [inline server/cmwell-data-tools-app/src/main/resources/logback.xml]: Do you want the bad-data to clutter the main log file? This is the current state...
The additivity should be false or am I missing something (maybe I am out of context and the log lines should be printed both on the main log file and also on the bad data one).

File: server/cmwell-dc/src/pack/logback.xml [hunk 3]
```diff
@@ -190,7 +214,6 @@
 
     <appender name="ASYNC_TOOLS_RED_LOG" class="ch.qos.logback.classic.AsyncAppender">
         <appender-ref ref="TOOLS_RED_LOG" />
-        <maxFlushTime>0</maxFlushTime>
```
Comment (e-orz) [inline server/cmwell-dc/src/pack/logback.xml]: same argument here. Why remove the maxFlushTime?

File: server/cmwell-dc/src/pack/logback.xml [hunk 4]
```diff
@@ -169,6 +169,30 @@
...
+        </rollingPolicy>
+        <encoder>
+            <pattern>%date{ISO8601} %-44.44([%thread]) %-5level %-36.36logger{36} - %msg%n</pattern>
+        </encoder>
+        <!-- false quadruples logging throughput -->
+        <immediateFlush>true</immediateFlush>
+    </appender>
+
+    <appender name="ASYNC_BAD_DATA" class="ch.qos.logback.classic.AsyncAppender">
+        <appender-ref ref="BAD_DATA" />
```
Comment (e-orz) [inline server/cmwell-dc/src/pack/logback.xml]: same argument - where is the maxFlushTime?

File: server/cmwell-dc/src/pack/logback.xml [hunk 5]
```diff
@@ -210,6 +233,10 @@
         <appender-ref ref="ASYNC_YELLOW_LOG" />
     </logger>
 
+    <logger name="bad-data" additivity="true">
```
Comment (e-orz) [inline server/cmwell-dc/src/pack/logback.xml]: Same argument as before. Do you want the log lines also on the main log file?

File: server/cmwell-data-tools/src/main/scala/cmwell/tools/data/utils/logging/DataToolsLogging.scala [hunk 6]
```diff
@@ -20,8 +20,8 @@ import org.slf4j.LoggerFactory
 case class LabelId(id: String)
 
 trait DataToolsLogging {
-  private[data] lazy val redLogger = LoggerFactory.getLogger("tools-red-logger")
-  private[data] lazy val badDataLogger = LoggerFactory.getLogger("bad-data")
+  private[data] lazy val redLogger = Logger(LoggerFactory.getLogger("tools-red-logger"))
+  private[data] lazy val badDataLogger = Logger(LoggerFactory.getLogger("bad-data"))
```
Comment (e-orz) [inline server/cmwell-data-tools/src/main/scala/cmwell/tools/data/utils/logging/DataToolsLogging.scala]: A suggestion: maybe you should follow the "convention"  and call the logger and the log file `tools-bad-data`

File: server/cmwell-data-tools-app/src/main/resources/logback.xml (hunk 1 above)
Comment (nicklester) [reply server/cmwell-data-tools-app/src/main/resources/logback.xml]: Agree done.

File: server/cmwell-data-tools-app/src/main/resources/logback.xml (hunk 2 above)
Comment (nicklester) [reply server/cmwell-data-tools-app/src/main/resources/logback.xml]: My mistake. Good spot. I was experimenting...

File: server/cmwell-dc/src/pack/logback.xml (hunk 3 above)
Comment (nicklester) [reply server/cmwell-dc/src/pack/logback.xml]: Agree. Donw

File: server/cmwell-dc/src/pack/logback.xml (hunk 4 above)
Comment (nicklester) [reply server/cmwell-dc/src/pack/logback.xml]: Done

File: server/cmwell-dc/src/pack/logback.xml (hunk 5 above)
Comment (nicklester) [reply server/cmwell-dc/src/pack/logback.xml]: Done

File: server/cmwell-data-tools/src/main/scala/cmwell/tools/data/utils/logging/DataToolsLogging.scala (hunk 6 above)
Comment (nicklester) [reply server/cmwell-data-tools/src/main/scala/cmwell/tools/data/utils/logging/DataToolsLogging.scala]: Yep - good idea. Done..
